### ISSUE TITLE
Converse method

### ISSUE THREAD

**Issue by JarbasAI** [chrison999/mycroft-skill-HelpMe#3]
User:  "Help me"
Mycroft:  "Say skill if you want help with a skill or commands if you want help with a command" (then waits)
User:  "skills"
Mycroft:  "Which skill do you want help with?" (then waits)
User:  "Weather"
Mycroft:  "The commands for the weather skills are..."

something like:

    def help_intent(self, message):
        self.speak("Say skill if you want help with a skill or commands if you want help with a command")
        self.expecting_answer = True
        self.counter = 0
        self.choice =  None
        self.asked = False

    def converse(self, transcript):
          utterance = transcript[0]
          # probably better to call another method inside this so it isnt very crowded
          if self.expecting_answer:
                   if "skill" in utterance:
                        self.expecting_answer = False
                        self.choice = "skill"
                        self.counter = 0
                   elif "command" in utterance:
                        self.expecting_answer = False
                        self.choice = "command"
                        self.counter = 0
                   elif counter >= 3 or "stop" in utterance:
                         self.speak("disabling help")
                         self.expecting_answer = False
                         self.choice = None
                         self.counter = 0
                         return False
                    else:
                         self.speak("try again")
                         self.counter += 1
                         self.choice = None
                         return True

             if self.choice is not None:
                      self.speak("what skill you want help for")
                      self.asked = True
                      self.choice = None
                      self.counter = 0
                      return True

             if asked:
                     # TODO grab skill from utterance
                     # TODO check if skill name is a skill or invalid
            
                     if self.choice == "skill":
                            self.speak("skill help")
                            self.asked = False
                            self.counter = 0
                            self.choice = None
                            return True
                     elif self.choice == "command":
                             self.speak("command help")
                             self.asked = False
                             self.counter = 0
                             self.choice = None
                             return True
                     elif counter >= 3 or "stop" in utterance:
                         self.speak("disabling help")
                         self.asked = False 
                         self.counter = 0
                         self.choice = None
                         return False
          return False

**Comment by chrison999** [comment]
Yes, that's the basic idea I had in mind.  The only thing I'd add is a third option for self choice - self.choice == "support" to trigger Mycroft replying with support info (website, email, etc.) for a skill (if available).
